### PR TITLE
[enhancement] enforce use of sklearn's ```check_is_fitted``` before checking oneDAL support

### PR DESCRIPTION
**Pull request by icfaust**
## Description

As a follow-up suggestion from #2109 , lets be consistent on where we call check_is_fitted.

This is stricter than sklearn's ```check_estimators_unfitted``` as it applies to anything that uses sklearnex's ```dispatch```.

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/tests/test_common.py [hunk 1]
```diff
@@ -365,7 +365,26 @@ def runtime_property_check(text, estimator, method):
...
+            onedal_support = "_onedal_cpu_supported"
+        elif "_onedal_gpu_supported" in text["funcs"]:
+            onedal_support = "_onedal_gpu_supported"
+        else:
+            pytest.skip(f"onedal dispatching not used in {estimator}.{method}")
+        # get location of _onedal_*_supported
+        idx = len(text["funcs"]) - 1 - text["funcs"][::-1].index(onedal_support)
+        validfuncs = text["funcs"][:idx]
+        assert (
+            "check_is_fitted" in validfuncs
```
Comment (david-cortes-intel) [inline sklearnex/tests/test_common.py:380]: Wouldn't it be better to use `mock` to ensure that the function is called in the test? Would also make it easier to understand what's going on during the tests.

Comment (david-cortes-intel) [reply sklearnex/tests/test_common.py:380]: I mean the built-in `mock` module, by adding something like `@mock.patch("sklearn.utils.validation.check_is_fitted")` on top of the test (+ an extra argument for it), and then checking the property `called` on the mocked function.

Comment (icfaust) [reply sklearnex/tests/test_common.py:380]: Very cool reference, it is definitely cleaner.

So your recommendation is as follows: refactor the general design tests procedure to instead of using trace to using a build-up fixture which collects the various spies, applies them, runs the estimator and method, and then stores the spies afterwards.

or run each estimator and method for this check_is_fitted test using a spy separate from the other procedures?

I guess I am trying to figure out how to get the ordering of various spies.

Comment (icfaust) [reply sklearnex/tests/test_common.py:380]: I guess I am worried that as we add more design rules its going to become a slippery slope back to just wanting a full trace...

Comment (david-cortes-intel) [reply sklearnex/tests/test_common.py:380]: I'm not sure about what would be the best way of arranging the tests.

As I see it, there's two places that use the "check that function was called" logic, which are for n_jobs and check_is_fitted. There's different ways that those could be added to the test, for example:
* Adding them as part of the test's body, by calling the "spy" initializer on the estimator fixture, which would need to move the `.fit` call to inside the body.
* Adding them as fixtures that would patch the function on initialize and unpatch at finalize, checking the function calls as part of the fixture finalization
* Adding them as a decorator to the test that would patch, unpatch, and assert.
* Turning the test into a class with an initializer that would fit the model and then the subsequent methods would test.

But that would leave the other design rules still as a list that gets checked inside the body, which might look inconsistent.

Not sure what's the best way here. My personal preference would be to have these two as fixtures and perhaps later on refactor the rest if needed.

Comment (samir-nasibli) [reply sklearnex/tests/test_common.py:380]: Very good discussion. Thank you @david-cortes-intel for the suggestions. These could refactoring could be done as a follow -up.
Currently I am good with this impl, since it's consistent with other this kind of the design rules check.

File: sklearnex/svm/svr.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def fit(self, X, y, sample_weight=None):
 
     @wrap_output_data
     def predict(self, X):
+        check_is_fitted(self)
```
Comment (samir-nasibli) [inline sklearnex/svm/svr.py:101]: many thanks for these alignments!

**Review by samir-nasibli — APPROVED** [review]
